### PR TITLE
[JENKINS-60926] - Fix agent installation as a service on Windows. (regression in 2.217)

### PR DESCRIPTION
**Pull request by jeffret-b**
See [JENKINS-60926](https://issues.jenkins-ci.org/browse/JENKINS-60926).

This corrects a minor regression introduced in Remoting 4.0 and Jenkins 2.217 (2020-01-23). A renaming broke agent Windows service installation. Combined with an upcoming release of slave-installer-module this will fix the issue. Supersedes #4464.

See the [Remoting changelog](https://github.com/jenkinsci/remoting/releases/tag/remoting-4.2). It contains this change plus a couple of minor cleanup PRs.

### Proposed changelog entries

* Fix agent installation as a service on Windows. (regression in 2.217) (issue 60926)
  * Remoting 4.2 changelog: https://github.com/jenkinsci/remoting/releases/tag/remoting-4.2
  * Agent Installer Module 1.7 changelog: https://github.com/jenkinsci/slave-installer-module/blob/master/CHANGELOG.md#17

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
Tests for this capability would be nice but would involve significant system setup.
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Comment by jeffret-b** [comment]
Need to get an incremental published so that I can make the corresponding change in slave-installer-module.

**Review by MarkEWaite — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
OK?

**Review by fcojfernandez — APPROVED** [review]

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
AFAICT we need a proper fix of https://github.com/jenkinsci/slave-installer-module/pull/7 to be integrated into this PR so that it addresses the issue. If we cannot make it in time, #4464 would be a way to go for the incoming weekly, and this PR would be a final fix for the next one

**Comment by jeffret-b** [comment]
There's an update to the slave-installer PR that works for both old and new versions of Remoting. Once we get that approved and released, we can update this PR with that version and this should be good to go.

**Comment by oleg-nenashev** [comment]
@jeffret-b https://github.com/jenkinsci/slave-installer-module/releases/tag/slave-installer-1.7 was released, working on the changelog. But you can update this PR

**Comment by oleg-nenashev** [comment]
https://github.com/jenkinsci/slave-installer-module/pull/8 for the changelog

**Review by oleg-nenashev — APPROVED** [review]
Thanks @jeffret-b ! +1 for getting this change integrated towards the next weekly.
I might be offline in next 24hrs due to the travel, and it would be great if somebody else merges it

**Comment by jeffret-b** [comment]
Updated the reference to slave-installer 4.7.

Thanks for the assistance in getting this through!

**Comment by jeffret-b** [comment]
> I might be offline in next 24hrs due to the travel, and it would be great if somebody else merges it

If you want, I can merge it after CI passes. Anything else we would need before merging?

**Comment by oleg-nenashev** [comment]
> If you want, I can merge it after CI passes. Anything else we would need before merging?

Would be nice to get few reviews, so I did a mass ping taking time constrains.
Also added the changelog links to the proposed changelog

**Review by oleg-nenashev — APPROVED** [review]
Reapproving, because I was re-requested after the group ping

**Review by timja — APPROVED** [review]

**Comment by jeffret-b** [comment]
We've got three approvals, which is probably good enough for this one. I'll go ahead and merge it on in so it will be ready for release.

**Pull request merged**
